### PR TITLE
Provider detection checkbox

### LINKED ISSUE #191: Implement provider detection
Usage protocols change depending on the provider that is used. Currently, CogniCrypt_SAST only supports one CrySL rule set of usage protocols and is provider-agnostic. Extend it to detect the used provider and select the corresponding set of CrySL rules. If no rules are available for the detected provider, make it default to the set of rules for the providers that ship with the JCA.

Consult the [JCA](https://docs.oracle.com/javase/9/security/java-cryptography-architecture-jca-reference-guide.htm#JSSEC-GUID-2BCFDD85-D533-4E6C-8CE9-29990DEB0190) and [JCA provider](https://docs.oracle.com/javase/9/security/howtoimplaprovider.htm#JSSEC-GUID-C485394F-08C9-4D35-A245-1B82CDDBC031) documentations for ways to invoke (specific) providers.

### PR DESCRIPTION
**Pull request by enriozuni**
# Provider detection analysis checkbox

This PR implements the provider detection checkbox in the CogniCrypt's preference page of Eclipse. The provider detection analysis functionality is imported from the CryptoAnalysis repository. The latest changes in provider detection analysis, as well as many provider detection related headless tests, are found in this [PR](https://github.com/CROSSINGTUD/CryptoAnalysis/pull/185) which is waiting to be merged to _develop_ branch of CryptoAnalysis.

Fixes #191 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Firstly, many headless tests regarding provider detection analysis were done and the PR is located [here](https://github.com/CROSSINGTUD/CryptoAnalysis/pull/185). Furthermore, various JUnit tests are carried out [here](https://github.com/CROSSINGTUD/CryptoAnalysis/tree/develop/CryptoAnalysis/src/test/java/tests/providerdetection). Lastly, numerous manual tests are made in inner Eclipse to prove the correctness of provider detection analysis. 

- [X] BragaCrypto headless tests on CryptoAnalyis repository
- [X] Provider detection JUnit tests on CryptoAnalysis repository
- [X] Various manual tests in inner Eclipse

**Test Configuration**:
* Eclipse Version: 2019-06
* Java Version: 1.8
* OS: Windows 10

# Checklist:

- [ ] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by kruegers** [comment]
The Jenkins server throws compiler errors. They look like they are related to CryptoAnalysis. Please investigate if you are buidling against the correct CryptoAnalysis version.

**Comment by enriozuni** [comment]
@kruegers yeah, that is right. I have made some changes in CryptoAnalysis regarding provider detection on my issue-135 branch. I have opened a PR for merging this branch into develop, but is currently being reviewed by Gökce.

**Comment by kruegers** [comment]
Isn't that PR about the headless test cases and not about the provider detection?

**Comment by enriozuni** [comment]
@kruegers yes, that one. I initially pushed the provider detection fixes to that branch. Now that I think of, it is better to clean some commits that are not related to the headless tests and open a new PR related to the provider detection fixes, so you can merge it to develop of CryptoAnalysis :)

**Comment by enriozuni** [comment]
@kruegers I opened a new PR with fixes regarding provider detection that are used in this PR. The PR can be found in CryptoAnalysis repo, under this [link](https://github.com/CROSSINGTUD/CryptoAnalysis/pull/213).

**Comment by kruegers** [comment]
Please pull from the develop branch. That should at least some of the compiler errors.

File: plugins/de.cognicrypt.core/.classpath [hunk 1]
```diff
@@ -8,7 +8,7 @@
 	<classpathentry exported="true" kind="lib" path="lib/commons-cli.jar"/>
 	<classpathentry exported="true" kind="lib" path="lib/commons-io.jar"/>
 	<classpathentry exported="true" kind="lib" path="lib/commons-lang3.jar"/>
-	<classpathentry exported="true" kind="lib" path="lib/CryptoAnalysis.jar"/>
```
Comment (kruegers) [inline plugins/de.cognicrypt.core/.classpath]: Please remove. That would result in errors on other machines.

**Review by kruegers — CHANGES_REQUESTED** [review]
LGTM, apart from that tiny thin. Please fix and PR can be merged. Thanks!

File: plugins/de.cognicrypt.core/.classpath (hunk 1 above)
Comment (enriozuni) [reply plugins/de.cognicrypt.core/.classpath]: I had forgotten it there. Just removed it.

**Comment by enriozuni** [comment]
The PR should be ready to be merged now

**Pull request merged**